### ISSUE TITLE
wrong start_line parameter in git link templates for module docstrings

### ISSUE THREAD

**Issue by fweisser** [pdoc3/pdoc#296]
### Expected Behavior

The generated "BROWSE GIT" link for a module should be generated with `start_line = 1` and `end_line` similarly with the correct last line number.

### Actual Behavior

It is generated with `start_line = 0` and `end_line` correspondingly with the last linenumber reduced by 1.

### Steps to Reproduce

1. Go to https://pdoc3.github.io/pdoc/doc/pdoc/cli.html
2. Click on "BROWSE GIT"-link beneath the module docstring. (for me it leads to https://github.com/pdoc3/pdoc/blob/0658ff01d8e218fcf67ee3f313bef6875c75f0ae/pdoc/cli.py#L0-L595)
3. Scroll to the bottom of the file and notice the last line that is not highlighted.

### Additional info

Every class, function, method or variable docstrings "BROWSE GIT"-link I have tried yet worked, so it seems to be specific to the module docstrings.

- pdoc version: 0.9.2

**Comment by kernc** [comment]
Thanks for the clear bug report. The issue seems to be the -1 here:
https://github.com/pdoc3/pdoc/blob/bec63077a03b2136e6982825d327073ed76cc20a/pdoc/html_helpers.py#L563-L564
Not sure why, though. Welcome an investigation/PR.

**Comment by fweisser** [comment]
Thank you for the quick response.

I did some further investigation and it seems that the real problem is an undocumented behavior of the `getsourcelines(object)` function in the `inspect` module.

The docstring for this function says:
> Return a list of source lines and starting line number for an object.

However, when looking through the source code of this function, I found:
https://github.com/python/cpython/blob/39a7578186d2f5eaa112a5854c46f84eae401522/Lib/inspect.py#L1011-L1016

This causes the function to return 0 as start_line for modules and for other objects the real line number (which is done with `lnum + 1`).

I also found a similar looking thing in the [findsource function](https://github.com/python/cpython/blob/39a7578186d2f5eaa112a5854c46f84eae401522/Lib/inspect.py#L837-L838) which is called by `getsourcelines()`so it seems to be "intended" behavior.

Therefore, I think this could easily be fixed by adding a check like (I just tried it locally and it seems to be working)
```python
if start_line == 0:
    start_line = 1
```
I can of course start a PR if you think that the above makes sense.
